bullet_train-outgoing_webhooks/app/models/concerns/webhooks/outgoing/delivery_support.rb

```diff
@@ -55,7 +64,19 @@ def still_attempting?
   end
 
   def failed?
-    !(delivered? || still_attempting?)
+    !delivered? && !still_attempting?
+  end
+
+  def not_attempted?
+    attempt_count.zero?
+  end
+
+  def attempts_schedule_period_elapsed?
+    created_at < max_attempts_period.ago
+  end
+
+  def failed_or_not_attempted_or_elapsed?
```

Comment:
It is used for the UI to mark deliveries as failed when there have been no attempts or the maximum period has elapsed.

Reply:
Just a note for @jagthedrummer: we do the first attempt quite quickly in the regular flow, but we can end up in a situation where we have not_attempted attempts and the endpoint deactivated in which case they won't be attempted anymore.